docs/tutorials/creating-your-first-substrate-chain/build.md

```diff
@@ -0,0 +1,180 @@
+|   +-- src
+|       |
+|       +-- lib.rs
+|       |
+|       +-- template.rs  <-- Edit this file
+|
++-- scripts
+|
++-- src
+|
++-- ...
+```
+
+You will see some pre-written code which acts as a template for a new runtime module. You can delete the contents of this file since we will start from scratch for full transparency.
+
+At a high level, the a Substrate Runtime Module can be broken down into 5 sections:
+
+```rust
+// 1. Imports
+use support::{decl_module, decl_storage, decl_event,...};
+
+// 2. Module Configuration
+pub trait Trait: system::Trait {...}
+
+// 3. Module Events
+decl_event! {...}
+
+// 4. Module Storage Items
+decl_storage! {...}
+
+// 5. Callable Module Functions
+decl_module! {...}
+```
+
+Things like events, storage, and callable functions should look familiar to you if you have done other blockchain development. We will show you what each of these components look like for a basic Proof Of Existence module.
+
+## Imports
+
+Since imports are pretty boring, you can start by copying this at the top of your empty `template.rs` file:
+
+```rust 
+use support::{decl_module, decl_storage, decl_event, ensure, StorageMap};
+use rstd::vec::Vec;
+use system::ensure_signed;
+```
+
+## Module Configuration
+
+For now, the only thing we will configure about our module is that it will emit some Events.
+
+```rust 
+/// The module's configuration trait.
+pub trait Trait: system::Trait {
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+```
+
+## Module Events
+
+Since we configured our module to emit events, let's go ahead and define that!
+
+```rust
+// This module's events.
+decl_event!(
+    pub enum Event<T> where AccountId = <T as system::Trait>::AccountId {
+        // Event emitted when a proof has been claimed.
+        ClaimCreated(AccountId, Vec<u8>),
+        // Event emitted when a claim is revoked by the owner.
+        ClaimRevoked(AccountId, Vec<u8>),
+    }
+);
+```
+
+Our module will only have two events:
+1. When a new proof is added to the blockchain.
+2. When a proof is removed.
+
+The events can contain some metadata, in this case, each event will also display who triggered the event (`AccountId`), and the proof data (`Vec<u8>`) that is being stored or removed.
+
+## Module Storage Items
+
+To add a new proof to the blockchain, we will simply store that proof in our module's storage. To store that value, we will create a [hash map](https://en.wikipedia.org/wiki/Hash_table) from the proof to the owner of that proof.
+
+```rust
+// This module's storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as PoeStorage {
+        // The storage item for our proofs.
+        // It maps a proof to the user who made the claim.
+        Proofs: map Vec<u8> => T::AccountId;
+    }
+}
+```
+
+If a proof has an owner, then we know that it has been claimed! Otherwise, the proof is still available to be claimed.
+
+## Callable Module Functions
```

Comment:
```suggestion
## Module Callable Functions
```

Reply:
Disagree with this change.

These are "Module Functions" which are "Callable"